tools/ports/__init__.py

```diff
@@ -63,6 +64,26 @@ def load_port(name):
       utils.exit_with_error('duplicate port variant: %s' % variant)
     port_variants[variant] = (port.name, extra_settings)
 
+  validate_port(port)
+
+
+def load_port_by_name(name):
+  port = __import__(name, globals(), level=1, fromlist=[None])
+  init_port(name, port)
+
+
+def load_port_by_path(path):
+  name = os.path.splitext(os.path.basename(path))[0]
+  if name in ports_by_name:
+    utils.exit_with_error(f'port path [{path}] is invalid: duplicate port name {name}')
```

Comment:
Can you use backticks to surround `path` and `name` here in the error message?

Reply:
Done. I also added backticks in other places